### PR TITLE
A proposed implementation for giving control on fog rendering. 

### PR DESCRIPTION
**Pull request by xtassin**
In some circumstances, we may want to enjoy the benefits of fog based tile loading optimization without the visual rendering of fog (ie. custom fog shader)

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @xtassin!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/fogRenderingControl/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Comment by ggetz** [comment]
Thanks @xtassin! 

@lilleyse Could you take a quick look and verify this is something that makes sense for us to support through the API before moving forward? Or is this an option that should live on the tileset or terrain itself?

**Comment by lilleyse** [comment]
I think it's fine since `enabled` is already a global property of `Fog` and `renderable` feels like similar scope.

File: Specs/Scene/GlobeSurfaceTileProviderSpec.js [hunk 1]
```diff
@@ -519,6 +519,35 @@ describe(
...
+        scene.camera.lookUp(1.2); // Horizon-view
+
+        return updateUntilDone(scene.globe).then(function () {
+          expect(scene).notToRender([0, 0, 0, 255]);
+
+          scene.fog.enabled = true;
+          scene.fog.renderable = false;
+          scene.fog.density = 1.0;
+          scene.fog.screenSpaceErrorFactor = 0.0;
+
```
Comment (ggetz) [inline Specs/Scene/GlobeSurfaceTileProviderSpec.js:544]: Thanks for adding the test! To make this a bit more robust, I would check to make sure that what renders when `renderable` is `false` is also not the color which renders when fog is enabled with `renderable` set to `true`. 


```suggestion
          scene.fog.enabled = true;
          scene.fog.density = 1.0;
          scene.fog.screenSpaceErrorFactor = 0.0;

          let result;
          expect(scene).toRenderAndCall(function (rgba) {
            result = rgba;
            expect(rgba).not.toEqual([0, 0, 0, 255]);
          });
          
          scene.fog.renderable = false;
          expect(scene).notToRender(result);
```

**Comment by ggetz** [comment]
@xtassin Could you please merge in the upstream changes from `main`? I'm having trouble checking on your branch to resolve the changes myself.

File: Specs/Scene/GlobeSurfaceTileProviderSpec.js [hunk 2]
```diff
@@ -519,6 +519,46 @@ describe(
...
+        return updateUntilDone(scene.globe).then(function () {
+          expect(scene).notToRender([0, 0, 0, 255]);
+
+          scene.fog.enabled = true;
+          scene.fog.renderable = false;
+          scene.fog.density = 1.0;
+          scene.fog.screenSpaceErrorFactor = 0.0;
+          scene.fog.enabled = true;
+          scene.fog.density = 1.0;
```
Comment (ggetz) [inline Specs/Scene/GlobeSurfaceTileProviderSpec.js:545]:
```suggestion
          scene.fog.density = 0.001;
```

Sorry, something went a bit weird with the suggestion. I think it was my fault.

**Comment by xtassin** [comment]
@ggetz Original PR is too old and I need to create a new Pull request in order to merge upstream: do you want me to do that?

**Comment by ggetz** [comment]
There shouldn't be a need to open a brand new pull request. You should be able to merge changes from the main branch of cesium into this one [using these instructions](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/CodeReviewGuide#merge), but let me know if you run into any issues. 

As a part of that process, you'll resolve the conflict in `CHANGES.md` by accepting both sets of changes. 

After merging, you should also be able to run `npm run prettier` to format all the code and commit the result, which should resolve the error which is showing up in the logs after running the checks on this branch.

**Comment by ggetz** [comment]
Awesome, thanks for the updates here @xtassin! Looking forward to getting your other pull requests in as well.

**Pull request merged**